clients/src/main/java/org/apache/kafka/clients/ApiVersion.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.kafka.clients;
+
+import org.apache.kafka.common.message.ApiVersionsResponseData.ApiVersionsResponseKey;
+import org.apache.kafka.common.protocol.ApiKeys;
+
+/**
+ * Represents the min version and max version of an api key.
+ *
+ * NOTE: This class is intended for INTERNAL usage only within Kafka.
```

Comment:
I assume this was based off of our `ApiVersions` class which has the same comment. Maybe some committer will know if there was a reason we didn't create an `kafka.clients.internals` module for such internal classes like we have for other packages

Reply:
Everything in `org.apache.kafka.clients` is actually internal.  None of those classes are public APIs.  However, the opposite is true for `org.apache.kafka.clients.producer`.  Everything in that namespace is a public API.  So we created `org.apache.kafka.clients.producer.internals` for the non-public producer stuff.

tl;dr: ApiVersion is fine in the `org.apache.kafka.clients` namespace.